libappstream-glib/as-app-desktop.c

```diff
@@ -456,32 +456,10 @@ as_app_parse_file_key_fallback_comment (AsApp *app,
 }
 
 gboolean
-as_app_parse_desktop_file (AsApp *app,
-			   const gchar *desktop_file,
-			   AsAppParseFlags flags,
-			   GError **error)
+as_app_parse_desktop_kf (AsApp *app, GKeyFile *kf, AsAppParseFlags flags, GError **error)
```

Comment:
@hughsie This function wasn't supposed to be a public api right? 
missing `static`

Reply:
Pardon me, but I don't see any call to `as_app_parse_desktop_kf` in [as-app.c](https://github.com/hughsie/appstream-glib/blob/master/libappstream-glib/as-app.c)
Am I missing something?

Reply:
My mistake, apologies! I thought this was about as_app_parse_desktop_data() -- i'll staicize the method now.

Reply:
Fixed in 1729786131ac68ec3f8dd100af484d23cb85ce65

Reply:
> Fixed in 1729786131ac68ec3f8dd100af484d23cb85ce65

Thanks. 

But I found yet another problem.
The symbols for `as-app-private.h` are actually getting exported. See in the [debian packaging](https://salsa.debian.org/pkgutopia-team/appstream-glib/blob/debian/0.7.16-1/debian/libappstream-glib8.symbols).

Asking around, I was advised that `G_GNUC_INTERNAL` should prevent exporting private-api as symbols. Not super-familiar with this stuff so can you please take a look what is going wrong? 
Doesn't RPM packaging also get all these private headers as symbols? :thinking: